CuteCom.pro

```diff
@@ -58,9 +58,12 @@ FORMS    += mainwindow.ui \
     searchpanel.ui
 
 RESOURCES += \
-    resources.qrc
+    resources.qrc \
+    styles/qdarkstyle/style.qrc
 
 DISTFILES += \
     qt.astylerc
 
+QMAKE_LFLAGS += -no-pie
```

Comment:
This line prevents from compiling when trying to build it with Qt Creator. (Unknown option)

Reply:
Hi! Please can you add more information about this? Because I'm compiling it right now using Qt Creator on Linux. Please explain me your test process so I can repeat it.

Looking the link I provided, it's Qt's gcc related flag. It is your build chain using a different compiler?

Anyway, I set this flag to fix a problem where executables are not recognized (double click from nemo launch the 'Open with ..' dialog instead of executing it, because is recognized as shared library instead executable, see the link i posted before for expanded information). Do you know a different fix?

Thanks

Reply:
I found this: https://stackoverflow.com/questions/45329372/ubuntu-recognizes-executable-as-shared-library-and-wont-run-it-by-clicking and tried to analyse in depth with CuteCom:

1. Only Debian-like distros will suffer this because gcc defaults on "-pie" flag ON, so "-no-pie" fix it
2. It seems only when using gcc and only at linking phase, others compilers are not affected
3. If set on .pro file, CMake still generates a library-like executable
4. If set on CMakeLists.txt, Qt Creator is not aware of this so a library-like executable is generated

Starting with Debian 9 and similar distros, default CuteCom executable is not correct: Can run inside Qt Creator and from command-line, but never by double-clicking it. I tested it compiling a fresh repository clone both from command line cmake+make and Qt Creator (default gcc kit ON). Reason is discussed on the link I provided.

To fix the incompatibilities you mentioned, I removed it from repository, but **how to deal with this?** A Debian user will not get a correctly composed executable, and needs to pass a flag to cmake / qt creator. **How to do this automatically?** Any idea?